compiler/src/test/java/com/squareup/anvil/compiler/dagger/InjectConstructorFactoryGeneratorTest.kt

```diff
@@ -1049,6 +1049,167 @@ public class InjectClass_Factory(
     }
   }
 
+  @Test fun `a factory class performs member injection on a super class from another module`() {
+
+    val otherModuleResult = compile(
+      """
+      package com.squareup.test
+
+      import javax.inject.Inject
+
+      abstract class Base {
+
+        @Inject
+        lateinit var base1: List<Int>
+
+        @Inject
+        lateinit var base2: List<String>
+      }
+      """
+    )
+
+    compile(
+      """
+      package com.squareup.test
+
+      import com.squareup.test.Base
+      import javax.inject.Inject
+      import javax.inject.Provider
+      import dagger.Lazy
+
+      class InjectClass @Inject constructor() : Base() {
+
+        @Inject
+        lateinit var name: String
+      }
+      """,
+      previousCompilationResult = otherModuleResult
```

Comment:
Again, @vRallev, this was such a great idea!

Chaining the compilations makes it so much easier to write these tests without creating one-off test classes elsewhere.